### ISSUE TITLE
Throw when class constructor is [[Call]]'d

### ISSUE THREAD

**Issue by caitp** [google/traceur-compiler#1848]
it's come to my attention that some people are writing classes which are later instantiated via `[[Call]]` (or, in angular's case, `fn.apply()`)

It's probably a good idea to at least provide a quick check here, so that people don't discover their code is buggy once they've migrated to browser implementations

**Comment by arv** [comment]
Yup. I'm not yet sure how to do this without degrading performance too much.

**Comment by caitp** [comment]
Babel is currently doing it with a simple

```js
var _classCallCheck = function(instance, Constructor) {
    if (!(instance instanceof Constructor)) {
        throw new TypeError("Cannot call a class as a function")
    }
};
```

--- which is not really spec compliant, but it's close enough for most uses, I think

**Comment by arv** [comment]
That seems good enough. Worth testing to see if that impacts the speed of the parser which creates a lot of ParseTree instances.

**Comment by sebmck** [comment]
@caitp No other way to do it unfortunately. :cry:
